core/admin/mailu/internal/views/auth.py

```diff
@@ -12,24 +12,26 @@
 def nginx_authentication():
     """ Main authentication endpoint for Nginx email server
     """
-    limiter = utils.limiter.get_limiter(app.config["AUTH_RATELIMIT"], "auth-ip")
-    client_ip = flask.request.headers["Client-Ip"]
-    if not limiter.test(client_ip):
-        response = flask.Response()
-        response.headers['Auth-Status'] = 'Authentication rate limit from one source exceeded'
-        response.headers['Auth-Error-Code'] = '451 4.3.2'
-        if int(flask.request.headers['Auth-Login-Attempt']) < 10:
-            response.headers['Auth-Wait'] = '3'
-        return response
     headers = nginx.handle_authentication(flask.request.headers)
     response = flask.Response()
+    ratelimit = app.config["AUTH_RATELIMIT"]
```

Comment:
Rate-limiting after the actual password check is a big no-no, I've proposed an alternative implementation in #1745